src/algebra/lie/basic.lean

```diff
@@ -544,64 +549,139 @@ end lie_algebra
 
 section lie_module
 
-variables (R : Type u) (L : Type v) [comm_ring R] [lie_ring L] [lie_algebra R L]
-variables (M  : Type v) [add_comm_group M] [module R M]
+variables (R : Type u) {L : Type v} {M  : Type w}
+variables [comm_ring R] [lie_ring L] [lie_algebra R L] [add_comm_group M] [module R M]
 
 /-- A Lie module is a module over a commutative ring, together with a linear action of a Lie
 algebra on this module, such that the Lie bracket acts as the commutator of endomorphisms. -/
-class lie_module extends linear_action R L M :=
-(lie_act : ∀ (l l' : L) (m : M), act ⁅l, l'⁆ m = act l (act l' m) - act l' (act l m))
+class lie_module (α : has_bracket L M) :=
+(add_lie : ∀ (x y : L) (m : M), ⁅x + y, m⁆ = ⁅x, m⁆ + ⁅y, m⁆)
+(lie_add : ∀ (x : L) (m n : M), ⁅x, m + n⁆ = ⁅x, m⁆ + ⁅x, n⁆)
+(smul_lie : ∀ (t : R) (x : L) (m : M), ⁅t • x, m⁆ = t • ⁅x, m⁆)
+(lie_smul : ∀ (t : R) (x : L) (m : M), ⁅x, t • m⁆ = t • ⁅x, m⁆)
+(lie_lie : ∀ (x y : L) (m : M), ⁅⁅x, y⁆, m⁆ = ⁅x, ⁅y, m⁆⁆ - ⁅y, ⁅x, m⁆⁆)
+
+@[simp] lemma lie_module_add_lie
+  (α : has_bracket L M) [lie_module R α] (x y : L) (m : M) :
+  ⁅x + y, m⁆ = ⁅x, m⁆ + ⁅y, m⁆ :=
+lie_module.add_lie R x y m
+
+@[simp] lemma lie_module_lie_add
```

Comment:
Would it be possible to call these simply `lie_add`, and also use them for brackets of `x y z : L`? Or does that have downsides?

Reply:
TLDR leave this with me, I'm probably going to refactor this substantially today/tomorrow. If you are inclined to read on, I'd be very grateful for any comments!

I believe it would; this is essentially what I was referring to in point 3 in my comments when I opened this PR.

I think the best way to achieve this would be to have the fundamental object not to be `lie_ring` but rather something like:
```lean
class lie_ring_module (L M : Type*) [add_comm_group L] [add_comm_group M] extends has_bracket L M
(add_lie : ∀ (x y : L) (m : M), ⁅x + y, m⁆ = ⁅x, m⁆ + ⁅y, m⁆)
(lie_add : ∀ (x : L) (m n : M), ⁅x, m + n⁆ = ⁅x, m⁆ + ⁅x, n⁆)
(lie_lie : ∀ (x y : L) (m : M), ⁅⁅x, y⁆, m⁆ = ⁅x, ⁅y, m⁆⁆ - ⁅y, ⁅x, m⁆⁆) -- or we could call this `jacobi`
```
and then define:
```lean
class lie_ring (L : Type*) [add_comm_group L] extends lie_ring_module L L
(lie_self : ∀ (x : L), ⁅x, x⁆ = 0)
```

I had considered and dismissed this idea but I am now returning to it because with the further unbundled approach, the equivalent of `lie_algebra_self_module` means we need two mathematically identical, but definitionally unequal, `has_bracket` instances on a Lie algebra. The associative case escapes this complication because one of them is `has_scalar` and the other is `has_mul` whereas I'm sort of forced to use `has_bracket` for both.

Reply:
This is now done! The original `lie_add` etc. lemmas that were specific to Lie algebras / rings have been deleted and replaced with their more general counterparts which apply to Lie (ring) modules. I verified by hand that each of the lemmas removed could be proved `by simp` with the new lemmas in place.